proto/v1/api/android.proto

```diff
@@ -0,0 +1,391 @@
+      // Indicates a topic that's not yet downloaded by the local client, but is available to be
+      // downloaded.
+      DownloadableTopic downloadable_topic = 1;
+
+      // Indicates a topic that's downloaded by the local client, but has a compatible update.
+      UpdatableTopic updatable_topic = 2;
+
+      // Indicates a topic that's both downloaded by the local client & full up-to-date. This mainly
+      // serves as a lightweight indicator to ensure that the server & client are properly in-sync
+      // with the state of the local client.
+      UpToDateTopic up_to_date_topic = 3;
+    }
+  }
+
+  // Represents a topic that's available to download and is not yet downloaded by the local client.
+  message DownloadableTopic {
+    option (org.oppia.proto.v1.versions.api_proto_version_type) = TOPIC_LIST_REQUEST_RESPONSE_PROTO_VERSION;
+
+    // The ID of the topic that's available to newly download. This is expected to be used to
+    // identify the local copy of the topic's summary in the event that the topic_summary field is
+    // empty. Note that content_version is not also included since topic_summary is only missing if
+    // the local client already has the latest, most up-to-date summary (including the corresponding
+    // content version).
+    string topic_id = 1;
+
+    // The full summary for the current topic. Note that this will be absent if the local client
+    // already has the latest summary downloaded as indicated by its download state context.
+    org.oppia.proto.v1.structure.DownloadableTopicSummary topic_summary = 2;
+
+    // The size, in bytes, needed to download the topic.
+    uint32 download_size_bytes = 3;
+  }
+
+  // Represents a topic that's available to be updated. Clients should use their existing topic
+  // summary when displaying this topic, and not update it until the user requests to download the
+  // topic. Future iterations of this API could include a preview summary here for clients to
+  // provide context on the changes to the topic to help inform users.
+  message UpdatableTopic {
+    option (org.oppia.proto.v1.versions.api_proto_version_type) = TOPIC_LIST_REQUEST_RESPONSE_PROTO_VERSION;
+
+    // The ID of the topic that's available to be updated. This is expected to correspond to an
+    // existing summary that the local client already has for the topic. In the event it doesn't, it
+    // may use the topic content API to retrieve the topic summary & treat this as a topic that's
+    // available to download.
+    string topic_id = 1;
+
+    // The new topic content version that may be updated to (as compared with the local summary for
+    // this topic which should have a smaller content version).
+    uint32 content_version = 2;
+
+    // The size, in bytes, required to download all missing pieces to update this topic.
+    uint32 update_size_bytes = 3;
+
+    // The precise list of structures that need to be updated in the local client in order to update
```

Comment:
Same concern here about confusion re what a "structure" is. Ditto elsewhere "structure" is used in a name/type.